### PR TITLE
doc: documentation for configurable cash register

### LINKED ISSUE #13: doc: document rest api in swagger
The REST api need to be documented using swagger's **YAML** format.

More information could be taken from:
http://swagger.io/
http://swagger.io/specification/
http://editor.swagger.io/#!/

### PR DESCRIPTION
**Pull request by gadjevb**
Added posprint.yml, this is a swagger documentation, thus allowing us to document the functionalities we implement. In the file is included the documentation for the functionality implemented in #14 .

Fixes #13

### CODE REVIEW

**Comment by gadjevb** [comment]
@mgenov what url should I put for the host field?

File: posprint.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          type: String
+        - name: description
+          in: request
+          description: Description for the device.
+          required: true
+          type: String
+      tags:
+        - Registers
+      responses:
+        200:
```
Comment (mgenov) [inline posprint.yml]: When device is already registered is not documented.

File: posprint.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Destination'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+definitions:
+  Product:
```
Comment (mgenov) [inline posprint.yml]: Device

File: posprint.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Product'
+  /v1/devices:
+    post:
+      summary: Registers device
+      description: |
+        Registers a single cash register device
+      parameters:
```
Comment (mgenov) [inline posprint.yml]: It's not params, it's a device registration request.

**Comment by gadjevb** [comment]
@mgenov made the required changes

File: docs/swagger/api.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  /v/devices:
+    post:
+      summary: Registers device
+      parameters:
+        - in: body
+          name: request
+          description: Registers a single cash register device
+          required: true
+          schema:
+            $ref: '#/definitions/Device'
```
Comment (mgenov) [inline docs/swagger/api.yaml]: DeviceRegistrationRequest

File: docs/swagger/api.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      tags:
+        - Registers
+      responses:
+        201:
+          description: The destination of the registered device
+          schema:
+            $ref: '#/definitions/Destination'
```
Comment (mgenov) [inline docs/swagger/api.yaml]: DeviceRegistrationResponse

File: docs/swagger/api.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
+swagger: '2.0'
+info:
+  title: Pos Print API
+  description: Application for cash registers operations
+  version: "1.0.0"
+host: yourname.telcong.com
+schemes:
+  - http
```
Comment (mgenov) [inline docs/swagger/api.yaml]: https

File: docs/swagger/api.yaml [hunk 7]
```diff
@@ -0,0 +1,74 @@
+swagger: '2.0'
+info:
+  title: Pos Print API
```
Comment (mgenov) [inline docs/swagger/api.yaml]: POS Print API

File: docs/swagger/api.yaml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  Destination:
+    type: object
+    properties:
+      destination:
+        type: string
+        description: Unique ip representing the endpoint of the device.
+  Error:
+    type: object
+    properties:
+      code:
```
Comment (mgenov) [inline docs/swagger/api.yaml]: You can add description with examples for each property.

**Comment by gadjevb** [comment]
@mgenov made the required changes!

File: docs/swagger/api.yaml (hunk 5 above)
Comment (mgenov) [inline docs/swagger/api.yaml]: .

**Comment by gadjevb** [comment]
@mgenov renamed Destination to DeviceRegistrationResponse

**Pull request merged**